### DISCUSSION TITLE
`MoveToBeginningOfLine` and `SelectToBeginningOfLine` work differently?

### DISCUSSION THREAD

**Discussion by osa1** [zed-industries/zed#21579] [Feature Requests]
I have these vim bindings:

```
[
  {
    "context": "Editor && !menu && vim_mode == normal && vim_operator == none && !VimWaiting",
    "bindings": {
      "H": "editor::MoveToBeginningOfLine",
      ...
    }
  },
  {
    "context": "Editor && vim_mode == visual && !VimWaiting",
    "bindings": {
      "H": "editor::SelectToBeginningOfLine",
      ...
    }
  }
]
```

The first one moves to the beginning of the line *after any indentation*, while the second one always moves to column 0.

The implementation of these commands use the same `indented_line_beginning` function: https://github.com/zed-industries/zed/blob/9487fffc55f47562b361c4f66273c73b8ef8041e/crates/editor/src/editor.rs#L8174-L8202

So I don't understand how can they behave differently. Is this a bug? Am I doing it wrong?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
